### ISSUE TITLE
indent rule may change within blocks of code

### ISSUE THREAD

**Issue by AndyOGo** [editorconfig-checker/editorconfig-checker.javascript#23]
First thanks for this awesome checker. I just started to play around with it and I enjoy it.

Considering an `.editorconfig` rc file, like:
```
indent_size = 2
```

This will give false positive errors for code snippets like:

```js
/*
 This comment block will cause an error, because it has just 1 space.
 Though it's a valid and intended convention/style.
 */
function foo() {}
```

Wrapping HTML for readability around multiple lines gives same errors for similar reasons:

```html
<div data-foo="1"
     data-bar="2" >
  <!-- Above will error because indentation is 5 spaces -->
</div>
```

I quickly looked at the current validation algorithm implemented here:
https://github.com/editorconfig-checker/editorconfig-checker.javascript/blob/master/src/validation/space/space-validator.js

Dealing with blocks, would need some contextual meta data being involved, like start of block (`/*`, `<div`), end of block (`*/`, `>`), maybe type of block, etc.

Did you find yourself in a situation like this before?

**Comment by mstruebing** [comment]
Are you sure that your first example is giving an error?
I thought I had taken care of. I even have this test which should exactly check this case: https://github.com/editorconfig-checker/editorconfig-checker.javascript/blob/master/src/validation/space/space-validator.test.js#L58

The second one should fail and I think that it is good so because the `.editorconfig` is saying otherwise even if the readability is far better than intending correctly. For that case, I think it would be better to implement that `// editorconfig-disable-line` modifier. 

WDYT?

**Comment by AndyOGo** [comment]
@mstruebing 
Thank you for your super fast answer.
I double checked the code from I created the first example.
Our code is omitting the preceding `* `, that's the issue - I corrected my example above.

Regarding the second case, we have also some other linters in use, where we allow and want this special indentation for readability reasons. We use it with React in our `JSX` files and for our HTML-like markup.

Personally I think this is very syntax/grammar related feature, it would be cool to have ignore configs like with eslint or stylelint. In that case I would even think of custom Regex for start and end of blocks.

**Comment by mstruebing** [comment]
Could we agree that I merge the other PR when you confirm that this is working as expected? 
Otherwise, it would mean too many code changes right now to implement that soonish because I currently only look at the current line of the file and didn't have any other context.

With PR #24 applied you could simply add a `// editorconfig-disable-line` at the end of the line - or if you are already in a comment just put it there at the end of the line. 

Great to see someone is really using this tool :+1:

**Comment by AndyOGo** [comment]
@mstruebing 
Thank you very much for your effort.

Indeed, I realised that the current approach is line based. Absolutely, changing this to a context-sensitive approach is much more work.

Personally I think having the option to disable it per single line is a great and quick improvement.

**Comment by mstruebing** [comment]
This is fixed for now.
